Source/WebCore/css/calc/CSSCalcTree+Simplification.h

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (C) 2024 Samuel Weinig <sam@webkit.org>
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. ``AS IS'' AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL APPLE INC. OR
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+ * EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ * PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+ * OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#pragma once
+
+#include "CSSCalcTree.h"
+#include "CSSToLengthConversionData.h"
+#include <optional>
+
+namespace WebCore {
+
+namespace Calculation {
+enum class Category : uint8_t;
+}
+
+namespace CSSCalc {
```

Comment:
The file names are CSSCalcTree+Foo but the namespace is just CSSCalc. Is there some logic behind this I'm missing or could we make the naming match? Similarly with CalculationTree+Foo

Reply:
The idea was just about grouping:

- All the CSSCalc related stuff goes in the CSSCalc namespace.  All the Calculation related stuff goes in the Calculation namespace.
- All the CSSCalc "tree" related stuff, goes in files prefixed with CSSCalcTree. All the Calculation "tree" related stuff, goes in files prefixed with CalculationTree.

I was kinda basing the name spacing on things like WebCore::Style and WebCore::Layout, where we use the namespace as the prefix for the files (e.g. StyleBuilder.h has WebCore::Style::Builder, LayoutTreeBuilder.h has WebCore::Layout::TreeBuilder etc.).

This breaks down a bit when each header doesn't correspond with a type. So, for the CSSCalcTree and CalculationTree, there are a few header/cpp files that are just about some algorithm being used on the CSSCalc::Tree / Calculation::Tree, so in those cases, I thought the +Foo.h/cpp notation was the most clear..